packages/lwc/types.d.ts

```diff
@@ -178,4 +180,45 @@ declare module 'lwc' {
      * @param config configuration object for the accessor
      */
     export function wire(getType: (config?: any) => any, config?: any): PropertyDecorator;
+
+    type WireConfigValue = Record<string, any>;
+    type WireContextValue = Record<string, any>;
+
+    interface WireAdapter {
+        update(config: WireConfigValue, context?: WireContextValue): void;
+        connect(): void;
+        disconnect(): void;
+    }
+
+    type WireDataCallback = (value: any) => void;
+    type WireAdapterSchemaValue = 'optional' | 'required';
+    type ContextValue = Record<string, any>;
+
+    interface ContextConsumer {
+        provide(newContext: ContextValue): void;
+    }
+
+    interface ContextProviderOptions {
+        consumerConnectedCallback: (consumer: ContextConsumer) => void;
+        consumerDisconnectedCallback?: (consumer: ContextConsumer) => void;
+    }
+
+    interface ContextProviderOptions {
+        consumerConnectedCallback: (consumer: ContextConsumer) => void;
+        consumerDisconnectedCallback?: (consumer: ContextConsumer) => void;
+    }
+
+    interface WireAdapterConstructor {
+        new (callback: WireDataCallback): WireAdapter;
+        configSchema?: Record<string, WireAdapterSchemaValue>;
+        contextSchema?: Record<string, WireAdapterSchemaValue>;
+    }
+
+    type createContextProviderReturnFn = (
```

Comment:
this is what's called a "Contextualizer" in the [wire reform](https://github.com/salesforce/lwc-rfcs/blob/master/text/0000-wire-reform.md#context-provider-for-wire-adapters). let's rename the type